cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -322,6 +330,58 @@ func (s *Server) Run(queue *Queue) {
 	}
 }
 
+// Update priority.json given new entries, and remove no longer tracked repos.
+// This doesn't simply write newPriorities because a transient getIndexOptions failure
+// would cause the associated repo to get deprioritized.
+func (s *Server) maybeUpdatePriorities(names []string, newPriorities map[string]float64) {
```

Comment:
idea: why can't we have the keys of `newPriorities` be same set as `names`? Then this function becomes much more straightforward, you just do `if !reflect.DeepEqual(priorities, newPriorities) { writeJSONFile(...) }`

I think the only change you need to do is always set priority, not only if priority is non-zero?

Reply:
I tried writing it like that, but it's possible for getIndexOptions to fail and timeout, which would cause the priorities to be removed and could cause a temporary drop in repo ranking. The more complicated path is necessary to add some error tolerance.

Reply:
Ok that makes sense. Sounds good.

> but it's possible for getIndexOptions to fail and timeout

Did you see this happening in practice?